controllers/azuremachine_reconciler.go

```diff
@@ -40,14 +40,14 @@ const (
 	DefaultBootstrapTokenTTL = 10 * time.Minute
 )
 
-// azureMachineService are list of services required by cluster actuator, easy to create a fake
+// azureMachineService is the group of services called by the AzureMachine controller
 type azureMachineService struct {
 	machineScope         *scope.MachineScope
 	clusterScope         *scope.ClusterScope
 	availabilityZonesSvc azure.GetterService
 	networkInterfacesSvc azure.Service
-	virtualMachinesSvc   azure.GetterService
-	disksSvc             azure.GetterService
+	virtualMachinesSvc   *virtualmachines.Service
```

Comment:
this is temporary to allow removing use of GetterService. In a follow up, `virtualMachinesSvc` will also become  an `azure.Service` after we remove all Get() calls from the controller (see #110 for more info).